code/modules/food_and_drinks/machinery/oven.dm

```diff
@@ -243,6 +252,43 @@
 	max_items = 6
 	biggest_w_class = WEIGHT_CLASS_BULKY
 
+/obj/item/plate/oven_tray/item_interaction(mob/living/user, obj/item/item,  list/modifiers, is_right_clicking)
+	. = ..()
+
+	if(isnull(item.atom_storage))
+		return .
+
+	if(!is_right_clicking)
```

Comment:
pretty big if block here, would be a better choice to early return `is_right_clicking`

ex
```
	if(is_right_clicking)
		var/obj/item/storage/tray = item

		for(var/obj/tray_item in contents)
			tray.atom_storage?.attempt_insert(tray_item, user, TRUE)

		return ITEM_INTERACT_SUCCESS
// rest of code can be de-indented
```